CS3219-AY2223S1/cs3219-project-ay2223s1-g12
# Password Hashing and Salting

Fixes #8 

Summary:
* Currently, password are stored in plain text - insecure way of storing passwords
* This PR will salt and hash passwords before storing to the database
* Implemented using bcrypt package
* This makes it more difficult for hackers to brute force users' passwords should the database be compromised
* Created _services.js_ file to handle logic processing of users
* Implemented function to verify password when logging in

How to test:

1. Install the relevant packages using npm i
2. Run 'npm run dev' to start the microservice
3. Make a post call on Postman Desktop to create a new user (http://localhost:8000/api/user )
4. Check mongodb atlas database contains the new user 

Possible Issues:
* Make sure your machine IP is whitelisted in mongoDB to be able to connect to the database (go onto the project page > security quickstart > add ur IP address at the very bottom of page)
